cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -731,28 +730,70 @@ public TSStatus executeNonQueryPlan(ExecutNonQueryReq request)
   public void syncLeaderWithConsistencyCheck(boolean isWriteRequest)
       throws CheckConsistencyException {
     if (isWriteRequest) {
-      if (!syncLeader()) {
-        throw CheckConsistencyException.CHECK_STRONG_CONSISTENCY_EXCEPTION;
-      }
+      syncLeader(new StrongCheckConsistency());
     } else {
-      switch (ClusterDescriptor.getInstance().getConfig().getConsistencyLevel()) {
+      switch (config.getConsistencyLevel()) {
         case STRONG_CONSISTENCY:
-          if (!syncLeader()) {
-            throw CheckConsistencyException.CHECK_STRONG_CONSISTENCY_EXCEPTION;
-          }
+          syncLeader(new StrongCheckConsistency());
           return;
         case MID_CONSISTENCY:
-          // do not care success or not
-          syncLeader();
+          // if leaderCommitId bigger than localAppliedId a value,
+          // will throw CHECK_MID_CONSISTENCY_EXCEPTION
+          syncLeader(
+              new CheckConsistency() {
+                @Override
+                public void postCheckConsistency(long leaderCommitId, long localAppliedId)
+                    throws CheckConsistencyException {
+                  if (leaderCommitId == Long.MAX_VALUE
+                      || leaderCommitId == 0
```

Comment:
0 or Long.MIN_VALUE?

Reply:
> 0 or Long.MIN_VALUE?

should be Long.MIN_VALUE,   commitIindexId is Long.Min_value for default..
I will alter..